pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -96,6 +96,7 @@ type Config struct {
 	SecretAccessKey string `hcl:"secret_access_key" json:"secret_access_key"`
 	Region          string `hcl:"region" json:"region"`
 	KeyMetadataFile string `hcl:"key_metadata_file" json:"key_metadata_file"`
+	KeyMetadata     string `hcl:"key_metadata" json:"key_metadata"`
```

Comment:
I'm not sure if `key_metadata` is really a config name that properly describes what the user needs to set here. This is actually originated by the fact that `key_metadata_file` may not be the best name either :)
To provide some context: `key_metadata_file` was chosen with the idea of pointing to a file that could have a bunch of metadata describing the key. What we have learned is that the file hasn't grown to have anything else rather than an identifier that's tied to the server.
Since we are adding a config that's related to this, we may take the opportunity to rename `key_metadata_file` and have names that better describe what the configs are for. Otherwise we would keep adding more confusion.
Maybe something like `key_identifier_file` and `key_identifier_value` are good options, or something along those lines. What do you think?
We should rename `key_metadata_file` in a backwards compatible way. This means that we can introduce the `key_identifier_file` setting while keeping `key_metadata_file` in 1.8.x, deprecate `key_metadata_file` in 1.9.0 and remove it in 1.10.

I know that this config name change may be out of scope for this PR. If you prefer to have that addressed separately, I'm happy to help with that.

What do you think?

Reply:
Yea, it wasn't a great name, I was just following the convention we'd started. I added this change.